framework/i18n/I18N.php

```diff
@@ -199,4 +199,108 @@ public function getMessageSource($category)
 
         throw new InvalidConfigException("Unable to locate message source for category '$category'.");
     }
+
+    /**
+     * Normalizes language string to "zh-Hant-CN", "zh-CN" or "CN" format ready
+     * to be used to specify translation language.
+     *
+     * @param string $language language string. Could be either:
+     *
+     * - Language tag according to https://www.rfc-editor.org/bcp/bcp47.txt
+     * - ICU locale string http://userguide.icu-project.org/locale
+     *
+     * @return string normalized language ID
+     *
+     * @since 2.0.4
+     */
+    public function normalizeLocale($locale)
+    {
+        if (!class_exists('\Locale', false)) {
+            return $this->fallbackNormalizeLocale($locale);
+        }
+
+        $data = [];
+
+        $parts = \Locale::parseLocale($locale);
+        if (isset($parts['language'])) {
+            $data[] = $parts['language'];
+        }
+        if (isset($parts['script'])) {
+            $data[] = $parts['script'];
+        }
+        if (isset($parts['region'])) {
+            $data[] = $parts['region'];
+        }
+
+        return implode('-', $data);
+    }
+
+    /**
+     * Normalizes language string to "zh-Hant-CN", "zh-CN" or "CN" format ready
+     * to be used to specify translation language. Doesn't require intl.
+     *
+     * @param string $language language string. Could be either:
+     *
+     * - Language tag according to https://www.rfc-editor.org/bcp/bcp47.txt
+     * - ICU locale string http://userguide.icu-project.org/locale
+     *
+     * @return string normalized language ID
+     *
+     * @since 2.0.4
+     */
+    protected function fallbackNormalizeLocale($locale)
```

Comment:
I think it should be private too, this is only a fallback if intl is not available which should not be exposed.